src/light.h

```diff
@@ -84,9 +84,7 @@ extern const u8 *light_decode_table;
 // 0 <= return value <= 255
 inline u8 decode_light(u8 light)
 {
-	if (light > LIGHT_MAX)
-		light = LIGHT_MAX;
-
+	assert(light <= LIGHT_SUN);
```

Comment:
Causes dark plantlike drawtype bug, needs to be
```
inline u8 decode_light(u8 light)
{
	if (light > LIGHT_SUN)
		light = LIGHT_SUN;

	return light_decode_table[light];
}
```

Reply:
Test debug build. I’m sure it will crash here with assertion failure. If so, the reason for the bug you describe is that some other code is wrong.

Reply:
Ok. Light level > 15 may be considered acceptable in some places, it may be intentional to have one check here in order to simplify code in some other places. However yes we should look at why plantlike is > 15, maybe that's a recent bug.

Reply:
@paramat The reason appears to be the fact that while `undiminish_light` refuses to increase light level if it is LIGHT_MAX already, it does increase it if it is greater (LIGHT_SUN in the case of plantlike, I suppose).

Reply:
Ok i guess it's best to fix that than cover up bugs in this PR.